plugin/src/main/java/ai/bryj/build/plugin/splashscreen/Splashscreen.kt

```diff
@@ -34,16 +34,20 @@ class Splashscreen(
             is SplashScreenConfigColor -> SplashScreenColorController(context, configuration)
         }
 
-        splashViewController.configure()
+        splashViewController?.configure()
 
-        splashViewController.onReadyToShowSplashScreen =
+        splashViewController?.onReadyToShowSplashScreen =
             accessPoint.nativeOverlaysController::addFullscreenOverlay
 
-        splashViewController.onReadyToHideSplashScreen =
+        splashViewController?.onReadyToHideSplashScreen =
             accessPoint.nativeOverlaysController::removeFullscreenOverlay
 
-        accessPoint.splashscreenListener = splashViewController.splashScreenListenerFilter
+        accessPoint.splashscreenListener = splashViewController?.splashScreenListenerFilter
+
+        accessPoint.androidLifecycleListener = splashViewController?.androidLifecycleListener
     }
 
-    override fun unregister() {}
+    override fun unregister() {
+        splashViewController = null
```

Comment:
Here you should release the the listeners on the accessPoint:
```suggestion
        accessPoint.splashscreenListener = null
        accessPoint.androidLifecycleListener = null
```